Integrate Atlas sensors code

utkarsh867 opened the pull request.

Milind220 committed:
Add atlas sensor library files and sensor driver code

Milind220 committed:
Integrate Atlas sensor classes.

Milind220 committed:
Fix faulty get data methods.

Milind220 committed:
Fix faulty _get_data method.

Milind220 committed:
Add docstring, comments.

Milind220 committed:
Minor changes.

Milind220 committed:
Fix remote I/O error with factory reset method.

Milind220 committed:
Fix initialise sensor method.

Milind220 committed:
Add ABC for AtlasI2C base class.

Milind220 committed:
Resolve TODO's.

Milind220 committed:
Minor change.

Milind220 committed:
Rename AtlasI2C.py to atlasI2C.py

Milind220 committed:
Add minor type hinting improvements.

Milind220 committed:
Change incorrect attribute names, add method calls

Milind220 committed:
Fix import

Milind220 committed:
Fix incorrect method name

Milind220 committed:
Add initialise sensor method

Milind220 committed:
Minor edit

utkarsh867 committed:
Add sensor annotations on image

utkarsh867 committed:
Fix bugs in sensor code

utkarsh867 commented:
Ready to test on a camera now. @Milind220 Let me know when you plan on doing that.

Milind220 commented:
Having lunch, should be back in about an hour - let’s test it then?

On Wed, 14 Jul 2021 at 2:34 PM, Utkarsh Goel ***@***.***>
wrote:

> Ready to test on a camera now. @Milind220 <https://github.com/Milind220>
> Let me know when you plan on doing that.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/oocam/camera/pull/18#issuecomment-879632072>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQNINJWGLNZNLHGLPPSPY53TXUVYPANCNFSM5AKXTF6A>
> .
>

utkarsh867 committed:
Add correction to the name of the keys in sensor readings

utkarsh867 committed:
Fix sensor reading for percentage oxygen

Milind220 committed:
Try to fix string to float conversion error

Milind220 committed:
Remove duplicate calibration attempts

Milind220 committed:
Try to fix list index error

Milind220 committed:
Add testfile for bug fixes

Milind220 committed:
Add additional checks for bugs

Milind220 committed:
Minor change to follow convention

Milind220 committed:
Add print statement to catch bug

Milind220 committed:
Change method from public to private

Milind220 committed:
Add TODO note for later improvement

utkarsh867 committed:
Fix import error and incorrect method name

Milind220 committed:
Add additional checks for bugs

utkarsh867 committed:
Merge branch 'atlas_sensors' of https://github.com/oocam/camera into atlas_sensors

Milind220 committed:
Add further checks for bugs

Milind220 committed:
Make minor changes

Milind220 committed:
Add further checks for list index bug

utkarsh867 committed:
Merge branch 'atlas_sensors' of https://github.com/oocam/camera into atlas_sensors

Milind220 committed:
Fix parameterEnableError for EC_Sensor

Along with correcting incorrect syntax, add more wait time to ensure
that previous calls are processed completely.

utkarsh867 committed:
Merge branch 'atlas_sensors' of https://github.com/oocam/camera into atlas_sensors

Milind220 committed:
AdD TODO notes for later dev + correct key names

To find the cause of the error of the data not being annotated to the
photos, the names of the keys have all been made uniform.

Seperately, TODO comments have been added where additional features need
to be added for the purpose of uniformity.

Milind220 committed:
Add more TODO comments

Milind220 committed:
Add checks to find missing photo annotation bug

The sensors were being read, but the readings aren't being annotated on
the photos at all. The checks will now show whether there is a prob with
data going in/out of the annotate_text_string method.

Milind220 committed:
Delete testfile.py

This file was required only to test the sensors, to find and solve the bug preventing them from being read properly.
This bug has since been found and fixed.

utkarsh867 committed:
Remove testfile.py

utkarsh867 committed:
Merge branch 'atlas_sensors' of https://github.com/oocam/camera into atlas_sensors

Milind220 committed:
Fix ascii encoding bug in camera.annotate_text

This fixes the bug that was preventing the sensor data from being
annotated onto the photos.

The built in camera.annotate_text method uses the ascii codec to encode
text, and write it onto the images. However, before this fix, the
'micro' symbol, which isn't a part of the ascii codec, was in the text.
This caused the error, and has now been corrected to 'micro' (the word).

Milind220 committed:
Merge branch 'atlas_sensors' of https://github.com/oocam/camera into atlas_sensors
Merge to reconcile minor changes between developement on the pi, and on
Milind's local.

Milind220 committed:
Remove testing print statements

Milind220 committed:
Remove testing print statements

Milind220 committed:
Remove testing print statements

Milind220 committed:
Remove _UNITS class attribute from all classes

The _UNITS class attribute was required during testing, but is no longer
required. The units of each measurement parameter can now be found in
the docstrings of each class.

Milind220 committed:
Remove get_header_row method from classes

This method was required during testing, no longer required.

Milind220 committed:
Refactor get_data methods for all classes

Print statements in the methods that were required for testing have been
removed.
Also, the methods now return -1 in the event of an exception, to be
consistent with the rest of the code.

Milind220 committed:
Remove print statements from DO_Sensor init

Milind220 committed:
Make minor changes

Milind220 committed:
Make minor change

Milind220 committed:
Add docstring for annotate_text_string method

Milind220 committed:
Remove unaccessed variable from capture_video

Milind220 committed:
Remove duplicate method definition

Milind220 committed:
Make a minor formatting change

Milind220 committed:
Fix syntax error

Milind220 committed:
Add type annotations

Milind220 committed:
Add exception handling classes for Atlas sensors

Milind220 committed:
Make a minor a change

Milind220 committed:
Add TODO for code review

Milind220 committed:
Add type annotations

Milind220 committed:
Fix missing import

Milind220 committed:
Resolve TODO comment

Milind220 committed:
Correct docstring in EC_Sensor

Milind220 committed:
Add missing units in annotate_text_string

The units for some of the readings from the sensors were missing. These
have now been added to provide clarity when looking at the readings.

Milind220 committed:
Remove extra line space

Milind220 committed:
Add TODO comment

Milind220 committed:
Remove unaccessed imports

Milind220 committed:
Add type annotations for init methods

Milind220 committed:
Correct type annotation for sensor_data

Milind220 committed:
Remove redundant imports

Milind220 committed:
Correct docstring for annotate_text_string

Milind220 committed:
Minorly reformat code

Milind220 committed:
Resolve TODO comment, remove print statement

Milind220 committed:
Remove unaccessed code

Milind220 committed:
Improve error/exception  message in _set_cal_data

Milind220 committed:
Refactor code a little

Milind220 committed:
Remove confusing comments

Milind220 committed:
Add Atlas sensor imports sensor package init

Milind220 committed:
Add type annotations

Milind220 committed:
Add TODO comment

utkarsh867 committed:
Fix write_sensor_data sensor object with timestamp
